config/crd/external-crds/vsphere.policy.vmware.com_storagepolicies.yaml

```diff
@@ -0,0 +1,111 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.16.1
+  name: storagepolicies.vsphere.policy.vmware.com
+spec:
+  group: vsphere.policy.vmware.com
+  names:
+    kind: StoragePolicy
+    listKind: StoragePolicyList
+    plural: storagepolicies
+    singular: storagepolicy
+  scope: Namespaced
+  versions:
+  - additionalPrinterColumns:
+    - jsonPath: .metadata.creationTimestamp
+      name: Age
+      type: date
+    name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: StoragePolicy is the Schema for the storagepolicies API.
+        properties:
+          apiVersion:
+            description: |-
+              APIVersion defines the versioned schema of this representation of an object.
+              Servers should convert recognized schemas to the latest internal value, and
+              may reject unrecognized values.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+            type: string
+          kind:
+            description: |-
+              Kind is a string value representing the REST resource this object represents.
+              Servers may infer this from the endpoint the client submits requests to.
+              Cannot be updated.
+              In CamelCase.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+            type: string
+          metadata:
+            type: object
+          spec:
+            properties:
+              id:
+                description: ID is the storage policy ID.
+                type: string
+            required:
+            - id
+            type: object
+          status:
+            properties:
+              datastores:
+                description: Datastores are the observed datastores selected by the
+                  storage policy.
+                items:
+                  properties:
+                    id:
+                      description: ID is the datastore's managed object ID.
+                      properties:
+                        objectID:
+                          description: ObjectID is the object's ID.
+                          type: string
+                        serverID:
+                          description: ServerID is the ID of the server to which the
+                            object belongs.
+                          type: string
+                      required:
+                      - objectID
+                      type: object
+                    type:
+                      description: Type is the observed type of datastore.
+                      enum:
+                      - VMFS
```

Comment:
Why only vmfs and vsan? Is there a way to distinguish between ESA and OSA? I remember DSM wanting this.

Reply:
Also, the reason I care about the difference between VSAN and VMFS is because multi-writer disks on VMFS require EZT and thin is supported on VSAN.